### ISSUE TITLE
crcmod

### ISSUE THREAD

**Issue by NegatioN** [conda-forge/google-cloud-sdk-feedstock#79]
Hi!

Is there any recommended way of install crcmod with this package?
I could only get it working by manually installing crcmod to pip afterwards, which doesn't seem optimal. Is it possible to include in the installation with some option, or parhaps as a default?

**Comment by scopatz** [comment]
What is crcmod, do you have more information?

**Comment by NegatioN** [comment]
Hi @scopatz sorry about that.
[crcmod](https://cloud.google.com/storage/docs/gsutil/addlhelp/CRC32CandInstallingcrcmod) is the library used by gsutil to verify composite object downloads. Without having a compiled C version of this installed, downloads of composite objects are significantly slower (and even raises by default).
Using this, we're able to speed up download/uploads by 20x ish.

I guess gsutil searches for the crcmod installation in a specific place, so the default one installed by `conda install -c conda-forge crcmod` is not detected.

**Comment by scopatz** [comment]
Hmm it seems like gsutil needs to be patched then to look for it elsewhere

**Comment by scopatz** [comment]
ie in the conda install dir

**Comment by NegatioN** [comment]
That might be correct. I'm not sure why it doesn't work out of the box when I look at the code: https://github.com/GoogleCloudPlatform/gsutil/blob/a93d02d37bdfb469dd3e60fe7cc70ce2f6d52bd8/gsutil.py#L111

It seems like it should just import the first crcmod on the `PYTHONPATH`, and I expect the conda libraries to be accessible.

Anyways, if this seems like a nofix, feel free to close it. And maybe I'll see if I ever get around to submitting a PR for gsutil or not.

**Comment by scopatz** [comment]
Seems like a real issue.  I am happy to keep this open; I just don't have the bandwidth to work on it myself :(

**Comment by BJWiley233** [comment]
for me it gets installed all the way in anaconda3/pkgs/crcmod-1.7-py37hc8dfbb8_1003/lib/python3.7/site-packages/crcmod directory.

**Comment by nehaljwani** [comment]
There is a feedstock for gsutil at https://github.com/conda-forge/gsutil-feedstock and it should be able to load crcmod (it's a runtime dependency already). One can simply uninstall the conda package `google-cloud-sdk` and install `gsutil` instead.

**Comment by enigmargs** [comment]
@nehaljwani I'm working on a HPC with a conda environment and installed crcmod using ` conda install -c conda-forge gsutil`. (version is 4.60). It still shows compiled **crcmod: False** under `gsutil version -l` and my downloads are getting stopped because of this. I don't have admin previlges to compile it from source. Any suggestion please?

**Comment by tdsmith** [comment]
I notice that when I `conda install -c conda-force crcmod` that the noarch package is being selected, which naturally must not be the compiled version. A compiled version exists at https://anaconda.org/conda-forge/crcmod and I'm not sure why it isn't being preferred. I am not sure whether it is possible to specify the architecture to install or to avoid the noarch flag.

**Comment by tdsmith** [comment]
Oh, no, the problem's even simpler; the crcmod feedstock just doesn't build the extension modules anymore. They're not in the built archive.

FWIW, explicitly specifying `conda install conda-forge/linux-64::crcmod` works fine in a Python 3.8 environment; there are no architecture-specific packages published for 3.9.

**Comment by nehaljwani** [comment]
This is fixed now:
```
$ ~/conda.exe create -yp ./dev -c conda-forge google-cloud-sdk
$ ~/conda.exe list -p ./dev google-cloud-sdk
# packages in environment at /path/to/dev:
#
# Name                    Version                   Build  Channel
google-cloud-sdk          357.0.0          py39hf3d152e_0    conda-forge
$ ./dev/bin/gsutil version -l | grep crcmod
compiled crcmod: False

$ ~/conda.exe install -p ./dev crcmod -c conda-forge
$ ~/conda.exe list -p ./dev crcmod
# packages in environment at /path/to/dev:
#
# Name                    Version                   Build  Channel
crcmod                    1.7             py39h3811e60_1006    conda-forge
$ ./dev/bin/gsutil version -l | grep crcmod
compiled crcmod: True
```

TL;DR: One can simply install the `crcmod` conda package from `conda-forge`. Just ensure that the version installed is `>=1.7`. If it is `1.7`, then the build number is `>=1006`
